src/language-js/print/ternary-old.js

```diff
@@ -0,0 +1,351 @@
+    // curly braces in an if statement.
+    const wrap = (doc) => [
+      ifBreak("("),
+      indent([softline, doc]),
+      softline,
+      ifBreak(")"),
+    ];
+
+    // The only things we don't wrap are:
+    // * Nested conditional expressions in alternates
+    // * null
+    // * undefined
+    const isNil = (node) =>
+      node.type === "NullLiteral" ||
+      (node.type === "Literal" && node.value === null) ||
+      (node.type === "Identifier" && node.name === "undefined");
+
+    parts.push(
+      " ? ",
+      isNil(consequentNode)
+        ? print(consequentNodePropertyName)
+        : wrap(print(consequentNodePropertyName)),
+      " : ",
+      alternateNode.type === node.type || isNil(alternateNode)
+        ? print(alternateNodePropertyName)
+        : wrap(print(alternateNodePropertyName)),
+    );
+  } else {
+    // normal mode
+    const part = [
+      line,
+      "? ",
+      consequentNode.type === node.type ? ifBreak("", "(") : "",
+      align(2, print(consequentNodePropertyName)),
+      consequentNode.type === node.type ? ifBreak("", ")") : "",
+      line,
+      ": ",
+      align(2, print(alternateNodePropertyName)),
+    ];
+    parts.push(
+      parent.type !== node.type ||
+        parent[alternateNodePropertyName] === node ||
+        isParentTest
+        ? part
+        : options.useTabs
+        ? dedent(indent(part))
+        : align(Math.max(0, options.tabWidth - 2), part),
+    );
+  }
+
+  // We want a whole chain of ConditionalExpressions to all
+  // break if any of them break. That means we should only group around the
+  // outer-most ConditionalExpression.
+  const shouldBreak = [
+    consequentNodePropertyName,
+    alternateNodePropertyName,
+    ...testNodePropertyNames,
+  ].some((property) =>
+    hasComment(
+      node[property],
+      (comment) =>
+        isBlockComment(comment) &&
+        hasNewlineInRange(
+          options.originalText,
+          locStart(comment),
+          locEnd(comment),
+        ),
+    ),
+  );
+  const maybeGroup = (doc) =>
+    parent === firstNonConditionalParent
+      ? group(doc, { shouldBreak })
+      : shouldBreak
+      ? [doc, breakParent]
+      : doc;
+
+  // Break the closing paren to keep the chain right after it:
+  // (a
+  //   ? b
+  //   : c
+  // ).call()
+  const breakClosingParen =
+    !jsxMode &&
+    (isMemberExpression(parent) ||
+      (parent.type === "NGPipeExpression" && parent.left === node)) &&
+    !parent.computed;
+
+  const shouldExtraIndent = shouldExtraIndentForConditionalExpression(path);
+
+  const result = maybeGroup([
+    printTernaryTest(path, options, print),
+    forceNoIndent ? parts : indent(parts),
+    isConditionalExpression && breakClosingParen && !shouldExtraIndent
+      ? softline
+      : "",
+  ]);
+
+  return isParentTest || shouldExtraIndent
+    ? group([indent([softline, result]), softline])
+    : result;
+}
+
+export { printTernaryOld };
```

Comment:
Can we rename this back as  `ternary.js` and name the new one as `experimental-ternary.js` or something else?

Reply:
My intent here is to make it as easy as possible to move to this when it's ready, and just remove all the old stuff. If we decide not to ship the new behavior, we can simply revert this PR. 

Does that work for you?

Reply:
But this makes PR hard to review. WDYT?

Reply:
Ah, the whole file was mostly rewritten, so I wouldn't recommend trying to compare it side-by-side anyway. 

If you'd like to do that, you can see a diff [here](https://github.com/prettier/prettier/compare/main...rattrayalex:prettier:postfix-ternaries-backup?expand=1#diff-e1193234d7131600f76f1feb77551740fc19959c03fee292e8e7f3dd23b06683) (it's a slightly outdated version, but hasn't changed much).

Reply:
if so, why don't send the rewrite first?

Reply:
Sorry, what do you mean? I originally intended to merge just the rewritten version and then add the option after, but Sosuke asked to include the option in this PR.

Reply:
If you'd like, I could rename `ternary.js` to `ternary-experimental.js` and leave everything else as-is, that might make it easier to review…

Reply:
If you are trying to check of `ternary-old.js` has changed since main, here is the diff:

```diff
diff --git a/src/language-js/print/ternary.js b/src/language-js/print/ternary-old.js
index 7f27026cc..e6a601719 100644
--- a/src/language-js/print/ternary.js
+++ b/src/language-js/print/ternary-old.js
@@ -189,7 +189,7 @@ function shouldExtraIndentForConditionalExpression(path) {
  * @param {Function} print - Print function to call recursively
  * @returns {Doc}
  */
-function printTernary(path, options, print) {
+function printTernaryOld(path, options, print) {
   const { node } = path;
   const isConditionalExpression = node.type === "ConditionalExpression";
   const consequentNodePropertyName = isConditionalExpression
@@ -348,4 +348,4 @@ function printTernary(path, options, print) {
     : result;
 }
 
-export { printTernary };
+export { printTernaryOld };
```

That is, the only thing which has changed is the name of the function.

I got this by running this command:
```
➜  git diff main:src/language-js/print/ternary.js postfix-ternaries:src/language
-js/print/ternary-old.js
```

Reply:
Sorry, I misunderstood, I thought you rewrote the old one.